nano/node/node.cpp

```diff
@@ -3320,10 +3318,16 @@ void nano::active_transactions::update_active_difficulty ()
 	}
 	assert (difficulty >= node.network_params.network.publish_threshold);
 	difficulty_cb.push_front (difficulty);
-	auto sum = std::accumulate (node.active.difficulty_cb.begin (), node.active.difficulty_cb.end (), uint128_t (0));
+	auto sum (std::accumulate (node.active.difficulty_cb.begin (), node.active.difficulty_cb.end (), uint128_t (0)));
 	difficulty = static_cast<uint64_t> (sum / difficulty_cb.size ());
 	assert (difficulty >= node.network_params.network.publish_threshold);
-	active_difficulty.store (difficulty);
+	atomic_active_difficulty.store (difficulty);
+}
+
+uint64_t nano::active_transactions::active_difficulty ()
+{
+	std::lock_guard<std::mutex> lock (mutex);
```

Comment:
Is this needed, as atomic_active_difficulty is atomic?

Reply:
this is to keep a timing issue from arising, for example: 
`active_difficulty = threshold;`
Thread 1
``` update_active_difficulty ()
//has lock and updates local active_difficulty calculation
active_difficulty = 20 ```
meanwhile
Thread 2
``` do_work_regeneration ()
if (active_difficulty > block_difficulty)
{
update_work_action (block)
}
```

Reply:
I see, seems like a comment in the code there about this is all that is needed, so the lock doesn't get removed by accident (since it locally looks like protecting an atomic)